data/tutorials/getting-started/3_01_ocaml_on_windows.md

```diff
@@ -6,8 +6,209 @@ description: >
+
+You will notice that the repository information fetching stage takes a while to
+complete. This is normal (for the moment), we advice our users to go get 
+themselves their favourite hot beverage while it runs.
+
+```shell-session
+> opam init
+No configuration file found, using built-in defaults.
+
+<><> Windows Developer Mode <><><><><><><><><><><><><><><><><><><><><><><><>  🐫
+opam does not require Developer Mode to be enabled on Windows, but it is
+recommended, in particular because it enables support for symlinks without
+requiring opam to be run elevated (which we do not recommend doing).
+
+More information on enabling Developer Mode may be obtained from
+https://learn.microsoft.com/en-gb/windows/apps/get-started/enable-your-device-for-development
+
+<><> Unix support infrastructure ><><><><><><><><><><><><><><><><><><><><><>  🐫
+
+opam and the OCaml ecosystem in general require various Unix tools in order to operate correctly. At present, this requires the installation of Cygwin to provide these tools.
+
+How should opam obtain Unix tools?
+> 1. Automatically create an internal Cygwin installation that will be managed by opam (recommended)
+  2. Use Cygwin installation found in C:\cygwin64
+  3. Use MSYS2 installation found in C:\msys64
+  4. Use another existing Cygwin/MSYS2 installation
+  5. Abort initialisation
+
+[1/2/3/4/5]
+Checking for available remotes: rsync and local, git.
+  - you won't be able to use mercurial repositories unless you install the hg command on your system.
+  - you won't be able to use darcs repositories unless you install the darcs command on your system.
+
+<><> Fetching repository information ><><><><><><><><><><><><><><><><><><><>  🐫
+[default] Initialised
+
+<><> Required setup - please read <><><><><><><><><><><><><><><><><><><><><>  🐫
+
+  In normal operation, opam only alters files within ~\AppData\Local\opam.
+
+  However, to best integrate with your system, some environment variables
+  should be set. When you want to access your opam installation, you will
+  need to run:
+
+    for /f "tokens=*" %i in ('opam env') do @%i
+
+  You can always re-run this setup with 'opam init' later.
+
+opam doesn't have any configuration options for cmd; you will have to run for /f "tokens=*" %i in ('opam env') do @%i
+whenever you change you current 'opam switch' or start a new terminal session. Alternatively, would you like to select a
+different shell? [y/n] n
+```
+
+One detail to address about having a fully functional opam installation is
+related to Git.
+
+On Windows, there are many ways to have a functioning Git installation. Opam will
+look for a compatible Git, and if none is found, it will prompt you with a set of
+options to install one. 
+
+```shell-session
+<><> Git ><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
+Cygwin Git is functional but can have credentials issues for private repositories, we recommend using:
+  - Install via 'winget install Git.Git'
```

Comment:
This would be good to know from the outset. https://ocaml.org/install#windows recommends doing
```
winget install Git.Git OCaml.opam
```